### PR TITLE
Changing pragmas to work with PIC 18F2685

### PR DESCRIPTION
**Pull request by mplewis**

### CODE REVIEW

File: masterChipSendSignal.c [hunk 1]
```diff
@@ -134,7 +134,8 @@ void Setup(DMXDevice *dmxDevice, char *dmxBuffer)
 
     SSPCON2 = 0b00000000;
 
-    //Baud = Fosc/(4*SSPADD+1) = ~114kHz when Fosc @ 12MHz
+    // Baud = Fosc / (4 * SSPADD + 1) = ~114kHz when Fosc @ 12MHz
+    // FIXME: Crystal @ 10 MHz, Fosc @ 40 MHz, change baud to match?
```
Comment (mplewis) [inline masterChipSendSignal.c]: @tylrtrmbl: this code references a target baud of 114 kHz, which is approximately the high-speed serial rate of 115200 Hz. Do we want this, or are we aiming for DMX baud of 250 kHz?

**Comment by taylortrimble** [comment]
Sorry, I saw this last night and didn't respond.

That is actually setup code for the I2C interface. There are three places where we have set up code:
- the `#pragma config` directives for setting the clock speed and crystal use
- the `Setup` function, which has the I2C setup code
- the `DMXSetup` function, which needs to have the 250 kbaud rate

I think in #44 we discussed using low-speed mode, so the baud rate should match that where your line comment is

**Comment by mplewis** [comment]
@tylrtrmbl, we should be good to go re: DMX baud rate. If I haven't missed anything, please confirm and ~~fire~~ merge at will.

**Comment by taylortrimble** [comment]
I like what I'm seeing so far. Thank you @mplewis!

File: DMXlib.c [hunk 2]
```diff
@@ -41,17 +41,34 @@
...
+    // RCSTA = 10010000 (SPEN = 1)
+    RCSTA = 0x90;
 
-    BAUDCONbits.BRG16 = 1; 	//Enable EUSART for 16-bit Asynchronous operation
+    // DMX baud rate is 250KHz
+    // If Fosc = 40 MHz, SPBRGH:SPBRG must be 39
     SPBRGH = 0;
-	
-    SPBRG = 31; 			//Baud rate is 250KHz for 32MHz Osc. freq.
+    SPBRG = 39;
```
Comment (taylortrimble) [inline DMXlib.c]: Let's get rid of this icky space

**Comment by CrasherCourse** [comment]
Functionality-wise I feel this code is ready to merge.

If we're still fixing comments, I saw a few places we could fix up a little that I'll list soon.

File: DMXlib.c [hunk 3]
```diff
@@ -88,26 +105,26 @@ void DMXReceive(DMXDevice *device)
...
                     break;
                 }
             case DMXWaitForData:
-                if (RCSTAbits.FERR) {	        //If a new framing error is detected (error or short frame)
-                    state = DMXWaitBreak;	// the rest of the frame is ignored and a new synchronization
-        	        break;                      //is attempted
-    	        }
-                if (PIR1bits.RCIF) {	        //Wait until a byte is correctly received
+                if (RCSTAbits.FERR) {            //If a new framing error is detected (error or short frame)
+                    state = DMXWaitBreak;    // the rest of the frame is ignored and a new synchronization
```
Comment (CrasherCourse) [inline DMXlib.c:104]: Perhaps moving "synchronization" to the next line would allow this comment to be properly tabbed.

**Comment by taylortrimble** [comment]
Eh, we have a style sweep coming up right after this anyway. I'm :apple: on this.

File: masterChipSendSignal.c [hunk 4]
```diff
@@ -134,7 +134,10 @@ void Setup(DMXDevice *dmxDevice, char *dmxBuffer)
 
     SSPCON2 = 0b00000000;
 
-    //Baud = Fosc/(4*SSPADD+1) = ~114kHz when Fosc @ 12MHz
+    // Baud = Fosc / (4 * SSPADD + 1) = ~114kHz when Fosc @ 12MHz
+    // Crystal @ 10 MHz, Fosc @ 40 MHz, change baud to match?
```
Comment (taylortrimble) [inline masterChipSendSignal.c:138]: Actually, one more thing: Let's remove these. In general, keep comments out of source code and on GitHub where we get "pinged!"

Comment (mplewis) [inline masterChipSendSignal.c:138]: Done.

**Pull request merged**